### PR TITLE
feat: system property to disable no delay provisioner cloud shuffle

### PR DESCRIPTION
**Pull request by cronik**
Add system property `io.jenkins.plugins.kubernetes.NoDelayProvisionerStrategy.disableCloudShuffle` to disable randomizing the cloud list. When disabled  clouds are search in the order they are defined making for more predicable pod template selection.

The use case for disabling is for multi-cloud environments where templates in both clouds share common labels and predictable cloud and pod template selection is desired. For example the first cloud is preferred and the second cloud is a backup, say for when the instance cap is hit.

### Testing done

Tested in live environment

```[tasklist]
### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

File: src/main/java/io/jenkins/plugins/kubernetes/NoDelayProvisionerStrategy.java [hunk 1]
```diff
@@ -56,17 +57,22 @@ public NodeProvisioner.StrategyDecision apply(NodeProvisioner.StrategyState stra
...
             Cloud.CloudState cloudState = new Cloud.CloudState(label, strategyState.getAdditionalPlannedCapacity());
+            searchClouds:
             for (Cloud cloud : jenkinsClouds) {
                 int workloadToProvision = currentDemand - availableCapacity;
                 if (!(cloud instanceof KubernetesCloud)) continue;
                 if (!cloud.canProvision(cloudState)) continue;
                 for (CloudProvisioningListener cl : CloudProvisioningListener.all()) {
                     if (cl.canProvision(cloud, cloudState, workloadToProvision) != null) {
-                        continue;
+                        continue searchClouds;
```
Comment (Vlatombe) [inline src/main/java/io/jenkins/plugins/kubernetes/NoDelayProvisionerStrategy.java:72]: For clarity, could you split this to a different PR?

**Review by Vlatombe — APPROVED** [review]

**Comment by jglick** [comment]
I struggle to understand why it would ever be shuffled. https://github.com/jenkinsci/kubernetes-plugin/pull/598#discussion_r326512506 suggests that the default behavior is intended as some sort of load balancing?

**Comment by scddev** [comment]
> I struggle to understand why it would ever be shuffled. [#598 (comment)](https://github.com/jenkinsci/kubernetes-plugin/pull/598#discussion_r326512506) suggests that the default behavior is intended as some sort of load balancing?

The problem arises as soon as you do not specify instance caps but rely on cpu/memory requests of the pods, so the cloud could spawn infinite pods, then every pod is created on the first cloud, the pod will end up in pending state.
